### PR TITLE
Public doc of backlog issues

### PR DESCRIPTION
**Pull request by jskeet**
These are issues that aren't on our radar at the moment.

The first pass has left quite a few not covered here, as I'd really
like to see them just get done.

Will close these issues after merging this commit.

(I'm assuming the `#` approach of linking to issues will work here - if not, I'll create a follow-up PR.)

### CODE REVIEW

File: BACKLOG.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Requested enhancements backlog
+
+This file is used to keep track of feature requests that we *may*
+want to come back to at some point, but aren't on our immediate
+radar. Using this single document is simpler than keeping all the
+corresponding issues open, but we'll reopen the issue.
```
Comment (chrisdunelm) [inline BACKLOG.md]: End of this sentence doesn't make sense

File: BACKLOG.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Static code analysis
+
+- Issue: #1709
+- Status: Further investigation required. We already have
+  some analysis in the form of custom Roslyn analyzers.
+
+## Metadata client failuer
```
Comment (chrisdunelm) [inline BACKLOG.md]: failuer failure ;)

File: BACKLOG.md (hunk 1 above)
Comment (jskeet) [reply BACKLOG.md]: Indeed - I'm sure it was longer than that before. Fixed.

File: BACKLOG.md (hunk 2 above)
Comment (jskeet) [reply BACKLOG.md]: Oops. Fixed.

**Pull request merged**